pkg/router/template/certmanager.go

```diff
@@ -124,16 +141,41 @@ func (cm *simpleCertificateManager) DeleteCertificatesForConfig(config *ServiceA
 			destCert, ok := config.Certificates[destCertKey]
 
 			if ok {
-				err := cm.w.DeleteCertificate(cm.cfg.caCertDir, destCert.ID)
-				if err != nil {
-					return err
-				}
+
+				destCertFile := certificateFile{CertDir: cm.cfg.caCertDir, ID: destCert.ID}
+				cm.deletedCertificates[destCertFile.Tag()] = destCertFile
 			}
 		}
 	}
 	return nil
 }
 
+// Commit applies any pending changes made to the certificateManager.
+func (cm *simpleCertificateManager) Commit() error {
+	// Deletion of certificates that are being referenced in backends or
+	// config is problematic in that the template router will not
+	// reload because the config is invalid, so we _do_ need to "stage"
+	// or commit the removals. Remove all the deleted certificates.
+	for _, certFile := range cm.deletedCertificates {
+		err := cm.w.DeleteCertificate(certFile.CertDir, certFile.ID)
+		if err != nil {
```

Comment:
I think we should log an error and ignore these certs. This will be same as what we do in cleanUpServiceAliasConfig() when DeleteCertificatesForConfig() fails.

Reply:
I agree.  There's no harm to having the stray cert around if the delete fails.  Let's log loudly and keep going.

Reply:
Will do.